libsplinter/src/admin/rest_api/actix/circuits.rs

```diff
@@ -396,4 +375,26 @@ mod tests {
 
         splinter_state
     }
+
+    fn run_rest_api_on_open_port(
+        resources: Vec<Resource>,
+    ) -> (RestApiShutdownHandle, std::thread::JoinHandle<()>, String) {
+        (10000..20000)
+            .find_map(|port| {
+                let bind_url = format!("127.0.0.1:{}", port);
+                let result = RestApiBuilder::new()
+                    .with_bind(&bind_url)
+                    .add_resources(resources.clone())
+                    .build()
+                    .expect("Failed to build REST API")
+                    .run();
+                match result {
+                    Ok((shutdown_handle, join_handle)) => {
+                        Some((shutdown_handle, join_handle, bind_url))
+                    }
+                    Err(_) => None,
```

Comment:
This should match the error which occurs when binding itself fails (as opposed to anything that can happen), so we stop the test with the error if its something unrelated to something already using the port.

Reply:
Added a commit to differentiate a bind error in the REST API's `run` method and used that in the tests.